pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -309,6 +309,13 @@ stdenv.mkDerivation rec {
 
     # Needed to find Mozilla runtime
     gappsWrapperArgs+=(--argv0 "$out/bin/.${binaryName}-wrapped")
+  '' + lib.optionalString (stdenv.isLinux && installAutoConfig) ''
+    # Prepare for autoconfig
+    mkdir -p "$out/lib/firefox/defaults/pref"
+    cat > "$out/lib/firefox/defaults/pref/autoconfig.js" <<EOF
+      pref("general.config.filename", "mozilla.cfg");
+      pref("general.config.obscure_value", 0);
```

Comment:
What's the point of this? Why these two options and why use autoconfig when this file won't ever change anyways (read-only nix store)?

Reply:
I think «prepare» is in terms of Nixpkgs-time, i.e. check that autoconfig works so that another change can actually introduce customisable autoconfig.

Reply:
That's the crazy way mozillas AutoConfig system works. The actual configuration file is added in the wrapper, but this file (autoconfig.js) has to be there in order for the AutoConfig machinery to do anything at all. Without it, firefox will just ignore the configuration file proper.
The first line tells firefox, that the name of the real configuration file is mozilla.cfg. This also has to be there, because some (enterprise) parties used it as a "lock down" mechanism, to prevent users from renaming it. In the same spirit, the configuration file (mozilla.cfg) is by default expected to be obfuscated by byte shifting. So the second line tells firefox that we use an non-obfuscated config file.
All of this is documented [here](https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig).

Reply:
Oh wow, can you add this link to the code?

Reply:
Done